pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -187,7 +223,45 @@ func (is *InstanceService) InstanceCreate(config *openstackconfigv1.OpenstackPro
 }
 
 func (is *InstanceService) InstanceDelete(id string) error {
-	return servers.Delete(is.computeClient, id).ExtractErr()
+	// get instance port id
+	allInterfaces, err := attachinterfaces.List(computeClient, id).allPages()
+	if err != nil {
+		return err
+	}
+	instanceInterfaces, err := attachinterfaces.ExtractInterfaces(allInterfaces)
+	if err != nil {
+		return err
+	}
+	portID := InstanceService[0].PortID
+
+	// delete instance
+	err := return servers.Delete(is.computeClient, id).ExtractErr()
+	if err != nil {
+		return err
+	}
+
+	// get and delete trunk
+	listOpts := trunks.ListOpts{
+		PortID: portID,
+	}
+	allTrunks, err := trunks.List(is.networkingClient, listOpts).AllPages()
+	if err != nil {
+		return err
+	}
+	trunkInfo, err := trunks.ExtractTrunks(allTrunks)
+	if err != nil {
+		return err
+	}
+	if len(trunkInfo) != 1 {
```

Comment:
By definition, there can't be more than one. So if len(trunkInfo) != 1, it is 0, so the error message should be fmt.Errorf("No trunk found for port ID: %s", portID)

Reply:
true, and actually, if there is no trunk it is ok too